core/eval.go

```diff
@@ -193,6 +197,23 @@ func evalSET(args []string) []byte {
 			}
 		case constants.KEEPTTL, constants.Keepttl:
 			keepttl = true
+		case "GET":
+			// Get the key from the hash table
+			obj := Get(key)
+
+			if XXoption {
+				if obj == nil {
+					return RespNIL
+				} else {
+					Put(key, NewObj(value, exDurationMs, oType, oEnc))
+				}
+			} else {
+				if obj == nil {
+					Put(key, NewObj(value, exDurationMs, oType, oEnc))
+				} else {
+					return RespNIL
+				}
```

Comment:
You still need to Put() the key even if it did not previously exist when using the GET option.

refer: https://redis.io/docs/latest/commands/set/

Reply:
@JyotinderSingh , but if the user passes `XX` or `NX` options along with the `GET` option, then also do we've to add the `Put()`?
Because, according to xx option, if the key doesn't exists, then it shouldn't be added, am I right?<br>
![image](https://github.com/user-attachments/assets/28963d06-6e78-4755-8643-0977d04131d6)